File: LOGICAL/SWIFT/leap-year.swift
import Foundation
print("Enter a year: ", terminator: "")
if let input = readLine(),let num = Int(input){
    let result = (num % 4 == 0) ?  "Leap Year" : "Not Leap Year"
    print("\(num) is \(result)")
} else {
    print("Invalid input. Please enter a valid number.")
}

File: LOGICAL/C/leap-year.c
#include <stdio.h>
#include <string.h>

int main()
{
    int num;

    printf("Enter a number: ");
    scanf("%d", &num);

    (num % 4 == 0) ? printf("The Year %d is a Leap Year.\n", num) : printf("The number %d is Not a Leap Year.\n", num);

    return 0;
}
// using ternay operator